tests/model/ManyManyThroughListTest.php

```diff
@@ -0,0 +1,275 @@
+<?php
+
+use SilverStripe\Dev\Debug;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\Dev\TestOnly;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use SilverStripe\ORM\ManyManyThroughList;
+use SilverStripe\ORM\Versioning\Versioned;
+
+class ManyManyThroughListTest extends SapphireTest
+{
+	protected static $fixture_file = 'ManyManyThroughListTest.yml';
+
+	protected $extraDataObjects = [
+		ManyManyThroughListTest_Item::class,
+		ManyManyThroughListTest_JoinObject::class,
+		ManyManyThroughListTest_Object::class,
+		ManyManyThroughListTest_VersionedItem::class,
+		ManyManyThroughListTest_VersionedJoinObject::class,
+		ManyManyThroughListTest_VersionedObject::class,
+	];
+
+	public function testSelectJoin() {
```

Comment:
Add a test for sorting by `SortOrder`. Sorting by a field on the join object is a pretty critical feature.